### ISSUE TITLE
Tray icon is missing since v5.0.0 (Linux)

### ISSUE THREAD

**Issue by fauust** [zulip/zulip-desktop#936]
---
Hi! Since I upgraded to version 5.0.0 the tray icon is missing from systray.

- **Operating System**:
  - [ ] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

WM: i3 version 4.18 + polybar 3.4.2
OS: Debian Buster (amd64)

- **Clear steps to reproduce the issue**:
Install latest Zulip desktop version from zulip repository (because of #912, I am not able to test with the Appimage version).

- **Relevant error messages and/or screenshots**:
Zulip is working perfectly but the tray icon is missing (I am getting notification with dunst).

**Comment by akashnimare** [comment]
@fauust did you install the app using our deb installer from the release page?

**Comment by fauust** [comment]
From https://zulipchat.com/help/desktop-app-install-guide
```bash
sudo apt-key adv --keyserver pool.sks-keyservers.net --recv 69AD12704E71A4803DCA3A682424BE5AE9BD10D9
echo "deb https://dl.bintray.com/zulip/debian/ stable main" | \
sudo tee -a /etc/apt/sources.list.d/zulip.list
sudo apt update
sudo apt install zulip
```

**Comment by devurandom** [comment]
I can confirm this behaviour for the Flatpak on multiple systems:

> (https://github.com/zulip/zulip-desktop/issues/910#issuecomment-609930694):
> On KDE Plasma, using the Flatpak, I never see any tray icon at all on these two systems (i.e. not even right after start of the application, which is a problem if `startMinimized=true` in `~/.config/Zulip/config/settings.json`):
> ```
> Operating System: Gentoo Linux 
> KDE Plasma Version: 5.18.4
> KDE Frameworks Version: 5.68.0
> Qt Version: 5.14.1
> Kernel Version: 5.5.14
> OS Type: 64-bit
> ```
> ```
> Operating System: Fedora 31
> KDE Plasma Version: 5.17.5
> KDE Frameworks Version: 5.67.0
> Qt Version: 5.13.2
> Kernel Version: 5.5.13-200.fc31.x86_64
> OS Type: 64-bit
> ```
> 
> See-Also: https://github.com/flathub/org.zulip.Zulip/issues/18

**Comment by abhigyank** [comment]
Hi @fauust @devurandom Chromium has changed how it shows tray icon, probably uses StatusNotifier tray icons now. [This comment](https://github.com/electron/electron/issues/21445#issuecomment-565710027) has the fix for some OSes, you can check other threads to support your OS if you can fine. If your OS doesn't support StatusNotifier tray, you might want to place an issue in that OS's tracker.
@akashnimare there isn't anything for us to do here. We can close.

**Comment by abhigyank** [comment]
@fauust For polybar, I found the upstream tracker here - https://github.com/polybar/polybar/issues/2010 , they haven't added support yet, but seems to be planning on doing it :)

**Comment by abhigyank** [comment]
@akashnimare I am putting a list up of StatusNotifier tray icons support status from different threads I found, so that it is easier for us to link - 

 * `KDE`: supported out-from-the-box.
 
 * `Xfce`: supported (`xfce4-statusnotifier-plugin` needs to be installed and then respective widget added to the panel). There is also https://gitlab.com/vala-panel-project/xfce4-sntray-plugin working plugin, alternative to `xfce4-statusnotifier-plugin`.
 * `Gnome Shell`: supported with the help of preinstalled [gnome-shell-extension-appindicator](https://github.com/ubuntu/gnome-shell-extension-appindicator) extension. Possible issues:
   
   * The tray/menu clicks might act flaky ([respective issue](https://github.com/electron/electron/issues/21576) placed).
   * The tray icon sporadically disappears, usually on unlocking the system from sleep/standby mode ([respective issue](https://github.com/electron/electron/issues/22443) placed).
 * `Cinnamon / Mint`: `"System Settings" -> "General" -> "Enable support for indicators"` flag needs to be enabled. Cinammon /Mint has has resolved [linuxmint/cinnamon#9176](https://github.com/linuxmint/cinnamon/issues/9176).
 * `polybar`: not supported ([respective issue](https://github.com/polybar/polybar/issues/2010) placed).

Some Linux users who still experience the tray icon issue with Electron-based apps might want to try https://git.sr.ht/~steef/snixembed thing.

**Comment by fauust** [comment]
@abhigyank, thanks for pointing me the snixembed project, that seems to work for the moment, see https://github.com/polybar/polybar/issues/2010#issuecomment-618284256.

**Comment by akashnimare** [comment]
@abhigyank awesome, we should document this somewhere (maybe in **limitations** section). cc @manavmehta

**Comment by devurandom** [comment]
> Hi @fauust @devurandom Chromium has changed how it shows tray icon, probably uses StatusNotifier tray icons now. [This comment](https://github.com/electron/electron/issues/21445#issuecomment-565710027) has the fix for some OSes, you can check other threads to support your OS if you can fine. If your OS doesn't support StatusNotifier tray, you might want to place an issue in that OS's tracker.

I am not sure I understand. Wayland environments do not support the X11 XEMBED extension for obvious reasons and never did. There is only SNI (Status Notifier Item) on Wayland. Swaybar supports it, but does not display a Zulip icon. Waybar also supports it and also does not display a Zulip icon. Other Flatpak applications' SNI (e.g. `com.borgbase.Vorta`) display their icons fine, without using `xembedsniproxy` or similar hacks.

**Comment by abhigyank** [comment]
@devurandom It is not not really a problem with Electron or Zulip. There isn't anything either can do here. If SNI support is present, I'd assume tray should work. I have read it works out of the box for KDE DE. You could create an upstream issue at Swaybar and Waybar. I did find a thread (probably on Arch) where people have reported that electron apps like Signal, Riot too don't have tray icon on Swaybar (issue with Swaybar maybe?). 
P.S. is Vorta an electron app?

**Comment by devurandom** [comment]
> @devurandom It is not not really a problem with Electron or Zulip. There isn't anything either can do here. If SNI support is present, I'd assume tray should work.

Is Zulip using a build of Electron that includes this change? https://github.com/electron/electron/pull/21904

> I have read it works out of the box for KDE DE.

I will try with KDE Plasma once I'm back on the computer running it.

> You could create an upstream issue at Swaybar and Waybar.

There do not seem to be any open ATM:
* Swaybar: https://github.com/swaywm/sway/issues?q=is%3Aissue+electron
* Waybar: https://github.com/Alexays/Waybar/issues?q=is%3Aissue+electron

Will open a report once I am back on the machine running Sway.

> I did find a thread (probably on Arch) where people have reported that electron apps like Signal, Riot too don't have tray icon on Swaybar (issue with Swaybar maybe?).

Link?

> P.S. is Vorta an electron app?

Does not look like it. Looks more like Python + Qt: https://github.com/flathub/com.borgbase.Vorta/blob/master/com.borgbase.Vorta.json

**Comment by abhigyank** [comment]
@devurandom Can you try in the latest release? Electron has reverted to the older way, so it should work as before.

**Comment by fauust** [comment]
Hi @abhigyank!
I can confirm that latest version is working, [snixembed](https://git.sr.ht/~steef/snixembed) is no more needed.

**Comment by akashnimare** [comment]
Alright. Closing this now. Thanks, everyone.

**Comment by devurandom** [comment]
> @devurandom Can you try in the latest release? Electron has reverted to the older way, so it should work as before.

Using KDE Plasma 5.19.2 (`xembedsniproxy` is running) and Flatpak 1.6.3 with Zulip 5.2.0, I do not see a tray icon. Permissions are at their defaults:
```desktop
[Context]
shared=network;ipc;
sockets=x11;wayland;pulseaudio;
filesystems=xdg-download;xdg-documents;xdg-pictures;

[Session Bus Policy]
org.kde.StatusNotifierWatcher=talk
org.freedesktop.Notifications=talk
```

**Comment by abhigyank** [comment]
@devurandom  please try on Zulip 5.3.0 .

**Comment by pskopek** [comment]
Zulip 5.3.0 works fine on Fedora 31 using GNOME and Topicons Plus.

**Comment by Moilleadoir** [comment]
> * Cinnamon / Mint: "System Settings" -> "General" -> "Enable support for indicators" flag needs to be enabled. Cinammon /Mint has has resolved linuxmint/cinnamon#9176.

This option no longer exists in System Settings.
